pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -695,21 +697,45 @@ func (qjm *XController) getAppWrapperCompletionStatus(caw *arbv1.AppWrapper) arb
 			}
 			unstruct.Object = blob.(map[string]interface{}) //set object to the content of the blob after Unmarshalling
 			name := ""
+			namespace := ""
 			if md, ok := unstruct.Object["metadata"]; ok {
 				metadata := md.(map[string]interface{})
 				if objectName, ok := metadata["name"]; ok {
 					name = objectName.(string)
 				}
+				if objectName, ok := metadata["namespace"]; ok {
+					namespace = objectName.(string)
+				}
 			}
+			_, gvk, err := unstructured.UnstructuredJSONScheme.Decode(objectName.Raw, nil, nil)
+			if err != nil {
+				klog.Errorf("[getAppWrapperCompletionStatus] Error unmarshalling, err=%#v", err)
+			}
+			
 			if len(name) == 0 {
 				klog.Warningf("[getAppWrapperCompletionStatus] object name not present for appwrapper: %v in namespace: %v", caw.Name, caw.Namespace)
 			}
 			klog.Infof("[getAppWrapperCompletionStatus] Checking items completed for appwrapper: %v in namespace: %v", caw.Name, caw.Namespace)
 
-			status := qjm.genericresources.IsItemCompleted(&genericItem, caw.Namespace, caw.Name, name)
+			status, condition := qjm.genericresources.IsItemCompleted(&genericItem, caw.Namespace, caw.Name, name)
 			if !status {
 				//early termination because a required item is not completed
-				return caw.Status.State
+				return caw.Status.State, completionStatus
+			} else {
+				_ = condition
```

Comment:
Q: what does this statement do?

Reply:
Ah, no longer need it. It just a way to silence go "unused var" error.